### PR TITLE
Honor usage of OidIsValid() macro

### PR DESCRIPTION
**Pull request by fabriziomello**
Postgres source code define the macro `OidIsValid()` to check if the Oid is valid or not (comparing against the `InvalidOid` type). See `src/include/c.h` in Postgres source three.

Changed all direct comparisons against `InvalidOid` for the `OidIsValid` call and add a coccinelle check to make sure the future changes will use it correctly.

### CODE REVIEW

File: coccinelle/oidisvalid.cocci [hunk 1]
```diff
@@ -0,0 +1,13 @@
+@@
+expression O;
+@@
+
+- (O != InvalidOid)
++ OidIsValid(O)
```
Comment (jnidzwetzki) [inline coccinelle/oidisvalid.cocci:6]: In the other Coccinelle checks, we also add a comment if the patch can be applied (see [this](https://github.com/timescale/timescaledb/blob/c06b647680dabac74cd027735ad7176f855c24a0/coccinelle/hash_create.cocci#L37) or [this](https://github.com/timescale/timescaledb/blob/c06b647680dabac74cd027735ad7176f855c24a0/coccinelle/hash_create.cocci#L37)). 

When the CI runs and one of the Coccinelle patches can be applied, the differences between the unpatched and the patched file are shown. The added comment makes it clearer for the user what the problem is.

**Review by jnidzwetzki — APPROVED** [review]
I have added a minor comment, the rest looks good to me.

**Review by svenklemm — APPROVED** [review]

File: coccinelle/oidisvalid.cocci (hunk 1 above)
Comment (svenklemm) [reply coccinelle/oidisvalid.cocci:6]: I agree please add a header to the file describing the check and a message when code does not conform so it's clear what the problem is

Comment (fabriziomello) [reply coccinelle/oidisvalid.cocci:19]: Added! Thanks!

**Pull request merged**